lib/heap_sort.rb

```diff
@@ -1,8 +1,22 @@
-
+require_relative "../lib/min_heap"
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1) 
 def heap_sort(list)
```

Comment:
👍   Since you are building a heap, the space complexity is O(n), and each add to the heap is O(log n) and you're doing it n times.  Then you have the same complexity for removing elements, so the time complexity is O(n log n)